pkg/chunked/compressor/compressor.go

```diff
@@ -218,3 +222,89 @@ func ZstdCompressor(r io.Writer, metadata map[string]string, level *int) (io.Wri
 
 	return zstdChunkedWriterWithLevel(r, metadata, *level)
 }
+
+type estarCompressor struct {
+	file             *os.File
+	writer           io.Writer
+	metadata         map[string]string
+	compressionLevel *int
+}
+
+func (c estarCompressor) Write(p []byte) (int, error) {
+	return c.file.Write(p)
+}
+
+func (c estarCompressor) Close() error {
+	defer c.file.Close()
+	st, err := c.file.Stat()
+	if err != nil {
+		return err
+	}
+	sr := io.NewSectionReader(c.file, 0, st.Size())
+
+	compressionLevel := pgzip.BestCompression
+	if c.compressionLevel != nil {
+		compressionLevel = *c.compressionLevel
+	}
+
+	blob, err := estargz.Build(sr, estargz.WithCompressionLevel(compressionLevel))
```

Comment:
estargz changes DiffID(uncompressed digest) of the layer because it adds TOC entry. Could we propagate the converted digest to the callers?

Reply:
It’s an extensive assumption of the current c/image/copy structure that compression/decompression/encryption/… doesn’t change DiffID values. Propagating this isn’t _impossible_ but it will touch a large part of the public API, probably triggering discussions about hiding some of the API, and so on…

Reply:
In particular we definitely don’t want to add `c/image/types.BlobInfo.DiffID`; that historical abuse of the transport API for internal needs of the copy implementation needs to stop at some point, so let’s stop now.